packages/genesys-spark-components/src/components/beta/gux-popover-beta/gux-popover.tsx

```diff
@@ -0,0 +1,321 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  JSX,
+  Listen,
+  Prop,
+  Watch,
+  State
+} from '@stencil/core';
+import {
+  autoUpdate,
+  computePosition,
+  arrow,
+  flip,
+  offset,
+  Placement,
+  shift
+} from '@floating-ui/dom';
+
+import { OnClickOutside } from '@utils/decorator/on-click-outside';
+import { trackComponent } from '@utils/tracking/usage';
+import { getSlot } from '@utils/dom/get-slot';
+import { findElementById } from '@utils/dom/find-element-by-id';
+
+/**
+ * @slot - popover content
+ * @slot title - Slot for popover title
+ */
+
+@Component({
+  styleUrl: 'gux-popover.scss',
+  tag: 'gux-popover-beta',
+  shadow: true
+})
+export class GuxPopover {
+  private popupElement: HTMLElement;
+  private arrowElement: HTMLDivElement;
+  private cleanupUpdatePosition: ReturnType<typeof autoUpdate>;
+
+  @Element()
+  private root: HTMLElement;
+
+  /**
+   * Indicates the id of the element the popover should anchor to
+   */
+  @Prop()
+  for!: string;
+
+  /**
+   * Indicate position of popover element arrow (follow floating ui placement attribute api)
+   */
+  @Prop()
+  position: Placement = 'bottom';
+
+  /**
+   * Indicate if the dismiss button is displayed
+   */
+  @Prop()
+  displayDismissButton: boolean;
+
+  /**
+   * Close popover when the user clicks outside of its bounds
+   */
+  @Prop()
+  closeOnClickOutside: boolean = false;
+
+  /**
+   * Controls hiding and showing the popover
+   */
+  @Prop({ mutable: true })
+  isOpen: boolean = false;
+
+  /**
+   * Fired when a user dismisses the popover
+   */
+  @Event()
+  guxdismiss: EventEmitter<void>;
+
+  @State()
+  private forElement: HTMLInputElement;
+
+  @Watch('for')
+  private updateForElement(): void {
+    this.forElement = this.getForElement();
+  }
+
+  @Listen('keydown')
+  onKeyDown(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'Escape':
+        this.dismiss();
+        break;
+    }
+  }
+
+  @Listen('focusout')
+  onFocusout(event: FocusEvent): void {
+    if (!this.closeOnClickOutside && this.displayDismissButton) {
+      return;
+    }
+
+    const focusIsOutsidePopover =
+      event.relatedTarget && !this.root.contains(event.relatedTarget as Node);
+
+    if (focusIsOutsidePopover) {
+      this.dismiss();
+    }
+  }
+
+  @OnClickOutside({ triggerEvents: 'mousedown' })
+  checkForClickOutside(event: MouseEvent) {
+    const clickPath = event.composedPath();
+    const forElement = findElementById(this.root, this.for);
+    const clickedForElement = clickPath.includes(forElement);
+
+    if (
+      (this.closeOnClickOutside || !this.displayDismissButton) &&
+      this.isOpen &&
+      !clickedForElement
+    ) {
+      this.dismiss();
+    }
+
+    if (clickedForElement) {
+      this.popupElement.showPopover();
```

Comment:
I also seen there is a `togglePopover` method which will show the popover if its hidden and vice versa. I was wondering would this be more beneficial then using `show/hidePopover()`. Not 100% sure what the pros and cons are. It does have some extra functionality that might be worth checking out 
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/togglePopover

Reply:
Fixed